tests/common/common.py

```diff
@@ -128,9 +129,19 @@ def trigger_and_retrieve(
             try:
                 log_list = self._get_logs(filter_str, ignore_protos)
                 return log_list
-            except (LogsNotFound, RpcError) as e:
+            except RpcError as e:
+                print(f"RPC error: {e}")
+                # most RpcErrors come from exceeding the reads per minute quota
+                # wait at least 60 seconds
+                # use a randomized backoff so parallel runs don't start up at 
+                # the same time again
+                sleep(random.randint(60, 300))
+                tries += 1
+            except LogsNotFound as e:
                 print("logs not found...")
-                sleep(5)
+                # logs may not have been fully ingested into Cloud Logging
+                # Wait another 10 seconds
+                sleep(10)
```

Comment:
I would implement exponentially growing number of seconds to sleep instead of the fixed one. API call quota is 1 minute based. So, the more chances are that (at least) last retry will cross the 1 minute boundary, more success rate to stay within the quota.

Reply:
> I think the main focus should be on LogsNotFound error. I think it is the main reason for quota exhaustion.

I'm not sure I agree with it being the more important part. The cause of the API exhaustion is multiple test instances running in parallel, all drawing from the same quota. Assuming we keep adding new environments, occasionally having bad scheduling and hitting quota issues is inevitable. I think it's more important how we handle and recover from a quota issue when it does happen.

I implemented the exponential wait though, so hopefully that will help too!